src/main.cpp

```diff
@@ -325,6 +328,9 @@ int main(int argc, char *argv[])
           return 1;
         }
         break;
+      case 'w':
```

Comment:
a `-w` flag that disables warning logging feels a bit counter intuitive.  A long option like `--no-warnings`  feels more descriptive to me, I don't think we need a short flag for this (you probably should fix the warnings)

Others compilers have the `-Wxxx` and `-Wno-xxx` flags, would be nice to keep  a similar option open in the future when we add that kind of granularity to our warnings.

Reply:
Yeah, `-w` feels strange to me, too and I got the idea since it seems gcc does so. I think `--no-warnings` is better.
> (you probably should fix the warnings)

You mean the failing tests in embedded builds?